Four pprobes

dzarda opened the pull request.

TVavrinec committed:
FourProbes

TVavrinec committed:
librarys

TVavrinec committed:
ochrany

TVavrinec committed:
YellowMotorEncoder_V2.0

New design with four sensors and xor to combine two signals to one.
Design is in KiCad.

TVavrinec committed:
3D_viewer

i add 3d model of motor and connector

TVavrinec committed:
.pro

i add missing soubor

TVavrinec committed:
.pro

i add missing soubor

TVavrinec committed:
Merge branch 'FourPprobes' of https://github.com/RoboticsBrno/RB0003-YellowMotorEncoder into FourPprobes

dzarda commented:
Schéma vypadá OK, jen bych měl připomínky k desce.

### Rozlošení sond

Hraju si tady s těmi sondami a snažím se to pochopit. Kolik že tam bude pólů?
![obrazek](https://user-images.githubusercontent.com/3955056/96262740-8129bb80-0fc2-11eb-9f1b-4d6ffcdd5934.png)


Tady vizualizace, když jsou sondy fázově 30° od sebe:
https://tinyurl.com/y2tqpmu2

Pak bych si taky dal bacha na jejich přesné pozice. Ideálně bych je všechny i stejně orotoval, ne že je tam naflákáš od oka:

![obrazek](https://user-images.githubusercontent.com/3955056/96265045-3d848100-0fc5-11eb-94a0-e3e6a9262b53.png)

dzarda commented:
Nemělo by to spíš vypadat takhle?

![obrazek](https://user-images.githubusercontent.com/3955056/96266997-9b19cd00-0fc7-11eb-904a-e0d5e4a20b05.png)

https://tinyurl.com/yyblc4up

TVavrinec commented:
počítám s 16 poly, přímo na desce sondy nemají stejné rozestupy co se pořadí v signálu týče tak jdou za sebou (v té verzi na kterou si koukal) A0, A1, B1, B0 (0, 45, 90, 135stupnů). Teď jsem to opravil na A0, B0, B1, A1.

TVavrinec committed:
rotate sensors

I changing position of sensors and rotating it in circle around hole for shaft of motor.

TVavrinec commented:
ty signály by měli vypadat spíš takhle
https://tinyurl.com/y3tnypnx

dzarda commented:
@TVavrinec To rozložení vypadá dobře, konkrétní úhly jsem nekontroloval; věřím, že je máš dobře. :) 

Dovolím si napsat něco k layoutu:

#### Tlumicí kondenzátory na motoru

Tady člověk chce držet co nejmenší smyčku, protože frekvenční obsah, které tím cestuje, je u DC motoru dost vysoko. Každé zvětšení smyčky zvětšuje magnetickou vazbu na okolí a následné EMI. Platí tady to stejné, jako u blokovacích kondů u IC. Mělo by to mít prioritu nad "pomalejšími" signály. Takhle vedou ty rychlé proudy:

![obrazek](https://user-images.githubusercontent.com/3955056/96929041-c9147b00-14b9-11eb-81ed-8c451a6e3ef9.png)

Lepší by mohlo být kondenzátorové cesty vést třeba takto:

![obrazek](https://user-images.githubusercontent.com/3955056/96929698-d41bdb00-14ba-11eb-9b1c-1132cebbf527.png)

#### Rozlití země

Souvisí s šířením rychlých proudů. Dobré je umožnit cestu zemi všude, kde je to možné (kde je logické, že by mohl chtít jít rychlý proud.)

![obrazek](https://user-images.githubusercontent.com/3955056/96930213-a4210780-14bb-11eb-92f6-17835d2d8912.png)
![obrazek](https://user-images.githubusercontent.com/3955056/96930266-b8650480-14bb-11eb-906c-1ae7737cf7ca.png)


#### Velikost prokovů

Vias by měly být stejně velké (pokud možno). Navíc 0,3mm díra s 0,5mm okružím je dost na hraně.

yaqwsx commented:
Doplním Jarka: 
- 0,3/0,5 mm via je nad schopnosti JLC PCB - deklarují minimální annular ring 0,13 mm, tzn. pro díru 0,3 bys měl mít pad velikosti 0,56, radějí více.

- nechceme pady motorů oválné místo kulatých? Bude se to výrazně lépe pájet

TVavrinec commented:
Oválné pedy rozhodně můžou bít.

Malé via opravím. Jenom otázka, je lepší mít všechny via malé nebo mít malí jen ti co se musí někam vlézt?

Co se té země týče, myslím si že ty problémy jsou způsobené velikostí desky nemyslím si že se to dá naskládat nějak řádově líp. Mimochodem bacha na ten prostřední ped, ten podle mě slouží docela dobře jako spojení ( myslím ten poslední obrázek, tam mi jde cesta která svojí tloušťkou zabraňuje průchodu země, ale ono de hned vedle tím pedem).
Jenak co mě napadá jako rychlé řešení té velké smyčky na prvním obrázku, co to tam jednoduše přerušit ať si chodí středem?

yaqwsx commented:
Pokud chceš malou desku, tak 4 vrstvy? Fakt je teď cena za 4 vrstvé desky pakatel. Zejména u takto malých destiček, kde jich do 10x10cm naskládáš hromadu.

TVavrinec commented:
jo to se mi líbí.

dzarda commented:
@TVavrinec Tu prostřední díru nechápu - proč je ukázaná, že má izolační mezeru od GND polygonu? Jestli je to spojený skrz to, tak cajk a 4 vrstvy snad nejsou třeba :)

Ad. řezání země, abych přiměl proud téct někde jinde, nebývá super nápad

Ad. velikost vias klasicky držím stejnou všude, pokud se nejedná třeba o nějaké vysokoproudy... Souvisí to taky s nutnou výměnou vrtáku při výrobě - JLC ti za to asi nic neúčtuje, ale je to krok navíc. Taky se na konzistentní desku líp dívá :)

TVavrinec commented:
ta díra nemá izolační mezeru to jsou jen thermáli, možná trochu atypicky ne svisle a vodorovně ale natočené o 45.

ok, mě právě podobnou věc doporučoval nějaký čas zpátky Kuba Štrait na B.B tak mě napadlo že by se to dalo použít.

ok, via dám všude malé. :)

dzarda commented:
Ahááá těch termálů jsem si nevšim :D ... Ty můžeš vydělat do pryč, když se tam nic nepájí.

TVavrinec commented:
no tak teoreticky by se tam pájel motor.
to má sloužit jako spojení země s pláštěm motoru aby člověk nemusel pájet extra kabel a jen ťukl pájkou.
